mmv1/templates/terraform/decoders/spanner_schedule_backup.go.erb

```diff
@@ -0,0 +1,15 @@
+config := meta.(*transport_tpg.Config)
```

Comment:
Could you briefly help me understand why the encoder/decoder is needed? On first look, it seems like we may be able to use built-in behavior instead.

Reply:
Sure: 
Encoder is needed to remove `instance`, `database` before making create/update calls because API does not expect those in body, also we want to initialize the name if it's not provided.
Decoder: Correctly set the resource ID, instance, database fields.
PreUpdate: to set updateMask for nested attributes.